### PR TITLE
Adjust task rating max value to 5

### PR DESCRIPTION
**Pull request by AleksiKytola**

### CODE REVIEW

File: server/tests/taskratings.test.ts [hunk 1]
```diff
@@ -95,6 +95,40 @@ describe('Test /roadmap/:roadmapId/tasks/:taskId/taskratings/ api', function ()
       );
       expect(before.body.length).to.equal(after.body.length);
     });
+    it('Should not add new taskrating with value outside min-max limits', async function () {
+      const firstRoadmapId = (await Roadmap.query().first()).id;
+      const taskId = (
+        await Task.query().where('roadmapId', firstRoadmapId).first()
+      ).id;
```
Comment (IidaKainu) [inline server/tests/taskratings.test.ts:102]:
```suggestion
      const { taskId, roadmapId: firstRoadmapId } = await getTestRatingData();
```

**Review by IidaKainu — CHANGES_REQUESTED** [review]
TaskHeatmap vaatinee vielä mukautusta

File: server/tests/taskratings.test.ts (hunk 1 above)
Comment (AleksiKytola) [reply server/tests/taskratings.test.ts]: Tests fixed & task heatmap roughly fixed

**Review by IidaKainu — CHANGES_REQUESTED** [review]
TaskHeatmap has a seemingly empty, transparent, row. I wonder what's going on? @AleksiKytola @juhoeerola

File: frontend/src/components/TaskHeatmap.module.scss [hunk 2]
```diff
@@ -23,8 +23,6 @@
   border: 1px solid $COLOR_SKY;
   padding-left: 16px;
   padding-top: 16px;
-  min-width: 386px;
-  min-height: 423px;
```
[the remark this answers is not in the document]
Comment (AleksiKytola) [reply frontend/src/components/TaskHeatmap.module.scss:27]: Re-added minimum size limits

File: frontend/src/components/TaskHeatmap.tsx [hunk 3]
```diff
@@ -22,15 +22,15 @@ export const TaskHeatmap = () => {
...
+  const frequencies = matrix2d({ rows: 5, cols: 5 });
   if (currentRoadmap) {
     currentRoadmap.tasks.map(ratingsSummaryByDimension).forEach((ratings) => {
       const value = ratings.get(TaskRatingDimension.BusinessValue);
       const work = ratings.get(TaskRatingDimension.RequiredWork);
       if (value && work) {
         const avgValue = Math.round(value.avg);
         const avgWork = Math.round(work.avg);
-        frequencies[10 - avgValue][avgWork] += 1;
+        frequencies[5 - avgValue][avgWork] += 1;
```
Comment (juhoeerola) [inline frontend/src/components/TaskHeatmap.tsx:33]: > TaskHeatmap has a seemingly empty, transparent, row

```suggestion
        frequencies[5 - avgValue][avgWork - 1] += 1;
```

Previously this was off by one logically but the grid had some extra room. Now the work can also go out of bounds of the grid, which creates mostly empty column.

**Review by IidaKainu — APPROVED** [review]
With the inclusion of Juho's suggestion the visualization makes sense 👍

**Comment by juhoeerola** [comment]
The axis labels could be slightly confusing, as the zero is really outside the grid. Changing that to 1, and moving those labels to the middle of (side of) the corresponding square *might* be clearer.

**Pull request merged**